alerting/notifier/channels/kafka.go

```diff
@@ -48,12 +54,36 @@ type KafkaNotifier struct {
 }
 
 type kafkaSettings struct {
-	Endpoint    string `json:"kafkaRestProxy,omitempty" yaml:"kafkaRestProxy,omitempty"`
-	Topic       string `json:"kafkaTopic,omitempty" yaml:"kafkaTopic,omitempty"`
-	Description string `json:"description,omitempty" yaml:"description,omitempty"`
-	Details     string `json:"details,omitempty" yaml:"details,omitempty"`
+	Endpoint       string `json:"kafkaRestProxy,omitempty" yaml:"kafkaRestProxy,omitempty"`
+	Topic          string `json:"kafkaTopic,omitempty" yaml:"kafkaTopic,omitempty"`
+	Description    string `json:"description,omitempty" yaml:"description,omitempty"`
+	Details        string `json:"details,omitempty" yaml:"details,omitempty"`
+	BasicAuth      bool   `json:"basicAuth,omitempty" yaml:"basicAuth,omitempty"`
```

Comment:
I think we do not need this field. I think validating that both user and password are specified would be enough

Reply:
Sounds good. I'll remove this.

Reply:
I have removed validation for these altogether. Give both, give either, don't give anything, I just pass it along. Does that work @yuri-tceretian ? Or would you prefer that I validate that if one is provided, the other should be too?

Reply:
From here: https://stackoverflow.com/a/74658311/4042839
it looks like username and/or password can be empty. So, I don't think we should validate that if one is provided, the other should be too.

@yuri-tceretian does that answer it? Can I mark this as resolved?

Reply:
Although I think this is not a good practice, I am ok.

Reply:
@yuri-tceretian I can add this if you prefer :)
But won't that be adding limitations on top of features allowed by Basic Auth? If Basic Auth allows username or password to be empty, then we should allow it as well, right?